qc_tests/streaks.py

```diff
@@ -93,102 +170,354 @@ def get_repeating_string_threshold(obs_var, config_dict, plots=False, diagnostic
-        threshold["Straight"] = th
+        get_repeating_streak_threshold(this_var, config_dict, plots=plots, diagnostics=diagnostics)
+        threshold = config_dict[f"STREAK-{this_var.name}"]["Straight"]
 
+    if threshold == utils.MDI:
+        # No threshold obtainable, no need to continue the test
+        return
+    
     # only process further if there is enough data
-    if len(this_var.data.compressed()) > 1:
-        repeated_string_lengths, grouped_diffs, strings = utils.prepare_data_repeating_string(this_var.data.compressed(), diff=0, plots=plots, diagnostics=diagnostics)
+    if len(this_var.data.compressed()) < 2:
+        # Escape if insufficient data
+        return
 
-        # above threshold
-        bad, = np.where(repeated_string_lengths >= threshold["Straight"])
+    repeated_streak_lengths, grouped_diffs, streaks = utils.prepare_data_repeating_streak(this_var.data.compressed(), diff=0, plots=plots, diagnostics=diagnostics)
 
-        # flag identified strings
-        for string in bad:
-            start = int(np.sum(grouped_diffs[:strings[string], 1]))
-            end = start + int(grouped_diffs[strings[string], 1]) + 1
+    # above threshold
+    bad, = np.where(repeated_streak_lengths >= threshold)
 
-            compressed_flags[start : end] = "K"
+    # flag identified streaks
+    for streak in bad:
+        start = int(np.sum(grouped_diffs[:streaks[streak], 1]))
+        end = start + int(grouped_diffs[streaks[streak], 1]) + 1
 
-            if plots:
-                plot_streak(times, this_var, start, end)
+        compressed_flags[start : end] = "K"
 
-        # undo compression and write into original object (the one with calm periods)
-        flags[this_var.data.mask == False] = compressed_flags
-        obs_var.flags = utils.insert_flags(obs_var.flags, flags)
+        if plots:
+            plot_streak(masked_times, this_var.data, obs_var.units, start, end)
+
+    # undo compression and write into original object (the one with calm periods)
+    flags[this_var.data.mask == False] = compressed_flags
+    obs_var.flags = utils.insert_flags(obs_var.flags, flags)
 
-    logger.info(f"Repeated Strings {this_var.name}")
+    logger.info(f"Repeated streaks {this_var.name}")
     logger.info(f"   Cumulative number of flags set: {len(np.where(flags != '')[0])}")
 
     return # repeating_value
 
 
 #************************************************************************
-def excess_repeating_value():
+def excess_repeating_value(obs_var: utils.Meteorological_Variable, times: np.array,
+                    config_dict: dict, 
+                    wind_speed: Optional[utils.Meteorological_Variable] = None,
+                    plots: bool = False, diagnostics: bool = False) -> None:
+    """
+    Flag years where more than expected fraction of data occurs in streaks,
+      but none/not many are long enough in themselves to trigger the repeating_value check
```

Comment:
This is still a bit confusing. 'excess_repeating_value' sounds like a test for overly frequent values but here you talk about short streaks repeating. So is this different from individual values that get erroneously repeated throughout the record?

Reply:
I've called this "annual string expectance" in HadISD - i.e. more observations in short strings than you'd expect (presuming most years are "good" data).  Would that be a better name?  I agree, there could be some confusion with the "frequent_values" check - though in that case there's no requirement for values to be in a chain of repeating ones.  Here, there has to be some repetition aspect - so lots of short strings, and more than expected (and excess number) - but none themselves long enough to set off the other test.